PianoFromAbove/GameState.cpp

```diff
@@ -1288,59 +1277,10 @@ void MainScreen::JumpTo(long long llStartTime, bool bUpdateGUI)
         m_iEndPos++;
 
     // Input position, iterators, tick
-    eventvec_t::const_iterator itOldProgramChange = m_itNextProgramChange;
     AdvanceIterators(llStartTime, true);
-    PlaySkippedEvents(itOldProgramChange);
     m_iStartTick = GetCurrentTick(m_llStartTime);
-
-    if (bUpdateGUI)
-    {
-        static PlaybackSettings& cPlayback = Config::GetConfig().GetPlaybackSettings();
         long long llNewPos = ((m_llStartTime - llFirstTime) * 1000) / (llLastTime - llFirstTime);
         cPlayback.SetPosition(static_cast<int>(llNewPos));
-    }
-}
-
-// Plays skipped program change and controller events. Only plays the one's needed.
-// Linear search assumes a small number of events in the file. Better than 128 maps :/
-void MainScreen::PlaySkippedEvents(eventvec_t::const_iterator itOldProgramChange)
-{
-    if (itOldProgramChange == m_itNextProgramChange)
-        return;
-
-    // Lookup tables to see if we've got an event for a given control or program. faster than map or hash_map.
-    bool aControl[16][128], aProgram[16];
-    memset(aControl, 0, sizeof(aControl));
-    memset(aProgram, 0, sizeof(aProgram));
-
-    // Go from one before the next to the beginning backwards. iterators are so verbose :/
-    vector< MIDIChannelEvent* > vControl;
-    eventvec_t::const_reverse_iterator itBegin = eventvec_t::const_reverse_iterator(m_itNextProgramChange);
-    eventvec_t::const_reverse_iterator itEnd = m_vProgramChange.rend();
-    if (itOldProgramChange < m_itNextProgramChange) itEnd = eventvec_t::const_reverse_iterator(itOldProgramChange);
-
-    for (eventvec_t::const_reverse_iterator it = itBegin; it != itEnd; ++it)
-    {
-        MIDIChannelEvent* pEvent = m_vEvents[it->second];
-        // Order matters because some events affect others, thus store for later use
-        if (pEvent->GetChannelEventType() == MIDIChannelEvent::Controller &&
-            !aControl[pEvent->GetChannel()][pEvent->GetParam1()])
-        {
-            aControl[pEvent->GetChannel()][pEvent->GetParam1()] = true;
-            vControl.push_back(m_vEvents[it->second]);
-        }
-        // Order doesn't matter. Just play as you go by.
-        else if (pEvent->GetChannelEventType() == MIDIChannelEvent::ProgramChange &&
-            !aProgram[pEvent->GetChannel()])
-        {
-            aProgram[pEvent->GetChannel()] = true;
-            m_OutDevice.PlayEvent(pEvent->GetEventCode(), pEvent->GetParam1(), pEvent->GetParam2());
-        }
-    }
-
-    // Finally play the controller events. vControl is in reverse time order
-    for (vector< MIDIChannelEvent* >::reverse_iterator it = vControl.rbegin(); it != vControl.rend(); ++it)
-        m_OutDevice.PlayEvent((*it)->GetEventCode(), (*it)->GetParam1(), (*it)->GetParam2());
```

Comment:
Why did this get removed? This breaks things like program changes being properly updated when seeking.